plugins/net_plugin/net_plugin.cpp

```diff
@@ -881,6 +880,16 @@ namespace eosio {
          
          return true;
       }
+      inline void setup_participant() {
+         if(my_impl->ssl_enabled) {
+            account_name participant = participant_name_ ? *participant_name_ : account_name{};
+            bool participating = my_impl->security_group.is_in_security_group(participant);
```

Comment:
need to take the connection_mtx here for calling is_in_security_group

Reply:
I checked usage of this mutex and it is used to lock connections vector to iterate or add/remove new connections.
why do we need it to lock before call to `is_in_security_group` ?